src/solver/misc/options.cpp

```diff
@@ -268,26 +268,18 @@ void checkAndCorrectSettingsAndOptions(Settings& settings, Data::StudyLoadOption
 
 void checkOrtoolsSolver(Data::StudyLoadOptions& options)
 {
-    std::string baseSolver = "sirius";
     if (options.ortoolsUsed)
     {
         const std::list<std::string> availableSolverList = getAvailableOrtoolsSolverName();
-        if (availableSolverList.empty())
-        {
-            throw Error::InvalidSolver(options.ortoolsSolver);
-        }
 
         // Check if solver is available
         bool found
           = (std::find(
                availableSolverList.begin(), availableSolverList.end(), options.ortoolsSolver)
              != availableSolverList.end());
-
         if (!found)
         {
-            logs.warning() << "Invalid ortools-solver option. Got '" << options.ortoolsSolver
-                           << "'. reset to " << baseSolver;
-            options.ortoolsSolver = baseSolver;
+            throw Error::InvalidSolver(options.ortoolsSolver, availableSolverList);
```

Comment:
what happened to throw an exception instead of using the base solver?

Reply:
It follows a discussion with @sylvlecl. According to him, falling back to Sirius without OR-Tools is not a desirable behavior since it can be misleading. The user thinks they're using XPRESS, but since it's not available (missing licence, misplaced env variables, etc.), Antares Solver silently uses Sirius with a warning in the logs.

The choice has been made to straight up refuse unavailable solvers.

Reply:
ok 👍🏽